### PR TITLE
Added convenience method to open a websocket to a server.

### PR DESCRIPTION
**Pull request by kumpelblase2**
Right now, you'd either have to do this yourself or monkey patch the http client to expose the socket. However, there should be something in the std lib to do that. Since I'm not too sure how people stand towards making socket a public property of the http client, I think this would make the most sense.

### CODE REVIEW

**Comment by kumpelblase2** [comment]
Not sure why an unrelated spec failed on the x64 arch :S

**Comment by asterite** [comment]
@kumpelblase2 Hey, we like this! But we think that the default path ("/") should be pretty rare. Can we make that path mandatory? It should be the second argument (`host, path, port = nil, ssl = false`) and that way we can also overload with `uri` that accepts a string or an URI. What do you think?

**Comment by kumpelblase2** [comment]
You're probably right with the path, I just made it optional for the sake of it being optional (so you can use it via named parameter) but I don't have any strong oppinion to keep it that way.

URI is actually a good idea, I'll do that. Question would be though, if only the correct schemas would be supported that way ("ws" and "wss" in that regard) or if "http" and "https" would be acceptable too. I would argue that it is OK but there might be a good reason not to do that.

**Comment by asterite** [comment]
We can support both protocols, I don't see a strong reason to fail in the case if http/https, and since it's WebSocket.open it's already clear what's happening.

**Comment by kumpelblase2** [comment]
Alright, I still added a note to make clear what happens when you use any other scheme.

File: src/http/web_socket.cr [hunk 1]
```diff
@@ -134,4 +141,57 @@ class HTTP::WebSocket
...
+  # ```
+  def self.open(uri : URI)
+    if host = uri.host
+      if path = uri.path
+        ssl = uri.scheme == "https" || uri.scheme == "wss"
+        return open(host, path, uri.port, ssl)
+      end
+    end
+
+    raise "No host or path specified which are required."
```
Comment (jhass) [inline src/http/web_socket.cr]: Let's make this an `ArgumentError`

File: src/http/web_socket.cr [hunk 2]
```diff
@@ -134,4 +141,57 @@ class HTTP::WebSocket
...
+    headers["Upgrade"] = "websocket"
+    headers["Sec-WebSocket-Version"] = VERSION.to_s
+    headers["Sec-WebSocket-Key"] = Base64.encode(StaticArray(UInt8, 16).new { rand(256).to_u8 })
+
+    path = "/" if path.empty?
+    handshake = HTTP::Request.new("GET", path, headers)
+    handshake.to_io(socket)
+    handshake_response = HTTP::Response.from_io(socket)
+    unless handshake_response.status_code == 101
+      raise "Handshake got denied. Status code was #{handshake_response.status_code}"
```
Comment (jhass) [inline src/http/web_socket.cr]: Anyone got an idea if there's a more specific exception class we can reuse here? `SocketError` perhaps?

Comment (kumpelblase2) [inline src/http/web_socket.cr]: `SocketError` probably seems the most reasonable.

**Comment by kumpelblase2** [comment]
Nice, thanks :)

**Pull request merged**